app/graphql/resolvers/concerns/collection.rb

```diff
@@ -7,21 +7,126 @@ module Collection
+        argument :sort_by, String, 'Sort by this searchable field', required: false
+        argument :sort_direction, Types::SortDirectionEnum, 'Sort direction', required: false
+
+        if has_taxonomix?
+          argument :location, String, required: false
+          argument :location_id, String, required: false
+          argument :organization, String, required: false
+          argument :organization_id, String, required: false
+        end
+
+        delegate :has_taxonomix?, to: :class
+      end
+
+      def resolve(**kwargs)
+        filters = filter_list(**kwargs)
+
+        base_scope = authorized_scope.all
+        filters.reduce(base_scope) { |scope, filter| filter.call(scope) }
+      end
+
+      private
+
+      def user
+        context[:current_user]
+      end
+
+      def model_class
+        self.class::MODEL_CLASS
+      end
+
+      def authorized_scope
+        return model_class unless model_class.respond_to?(:authorized)
+
+        permission = model_class.find_permission_name(:view)
+        model_class.authorized_as(user, permission, model_class)
+      end
+
+      def search_options(search:, sort_by:, sort_direction:)
+        (sort_direction.to_s.upcase == 'DESC') ? 'DESC' : 'ASC'
+        search_options = [search]
+        if sort_by.present?
+          search_options << { order: "#{sort_by} #{sort_direction}".strip }
+        end
+        search_options
+      end
+
+      def filter_list(search: nil, sort_by: nil, sort_direction: 'ASC',
+                      first: nil, skip: nil,
+                      location: nil, location_id: nil,
+                      organization: nil, organization_id: nil)
+        filters = []
+
+        if search.present? || sort_by.present?
+          filters << search_and_sort_filter(search: search, sort_by: sort_by, sort_direction: sort_direction)
+        end
+
+        if has_taxonomix?
+          taxonomy_ids = taxonomy_ids_for_filter(location_id: location_id, organization_id: organization_id,
+                                                 location: location, organization: organization)
+          filters << taxonomy_id_filter(taxonomy_ids) if taxonomy_ids.any?
+        end
+
+        filters
+      end
+
+      def search_and_sort_filter(search:, sort_by:, sort_direction:)
+        lambda do |scope|
+          begin
+            scope.search_for(*search_options(search: search, sort_by: sort_by, sort_direction: sort_direction))
+          rescue ScopedSearch::QueryNotSupported => error
+            raise GraphQL::ExecutionError.new(message: error.message)
+          end
+        end
       end
 
-      def resolve(search: nil, order: nil, order_by: nil)
-        params = {
-          order_by: order_by,
-          order: order,
-          search: search
-        }
+      def taxonomy_id_filter(taxonomy_ids)
+        lambda do |scope|
+          scope.joins(:taxable_taxonomies)
+            .where(taxable_taxonomies: {taxonomy_id: taxonomy_ids}).distinct
```

Comment:
I believe the name of `scope_by_taxable_ids` is misleading. It's used to apply a filter on the actual resource (and not join the resource with the taxonomy table and filter the taxonomies by the given ids).
I haven't found another suitable method, though. We could add a method to `Taxonomix`, but I'm not sure if that'll lead to cleaner code.

Reply:
Taxonomix could definitely be improved :) I think that might be because this join could be very painful on large tables as taxable_taxonomies is usually huge.